File: docs/guides/java/src/test/java/com/algolia/mcmSearchWithout.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;
import java.util.List;

public class mcmSearchWithout {

  private static String getAppIDFor(String user) {
    return ""; // Implement your own logic here
  }

  private static String getIndexingApiKeyFor(String user) {
    return ""; // Implement your own logic here
  }

  public static void main(String[] args) throws Exception {
    // Fetch from your own data storage and with your own code
    // the associated application ID and API key for this user
    String appID = getAppIDFor("user42");
    String apiKey = getIndexingApiKeyFor("user42");

    try (SearchClient client = new SearchClient(appID, apiKey)) {
      SearchParams searchParams = new SearchParamsObject()
        .setQuery("<YOUR_SEARCH_QUERY>")
        .setFacetFilters(FacetFilters.of(List.of(FacetFilters.of("user:user42"), FacetFilters.of("user:public"))));

      client.searchSingleIndex("<YOUR_INDEX_NAME>", searchParams, Hit.class);
    } catch (Exception e) {
      System.out.println("An error occurred: " + e.getMessage());
    }
  }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/mcmSearchWithout.kt
import kotlinx.serialization.json.JsonObject

import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.transport.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*

suspend fun mcmSearchWithout() {
  val getAppIDFor: (String) -> String = {
    "" // Implement your own logic here
  }
  val getIndexingApiKeyFor: (String) -> String = {
    "" // Implement your own logic here
  }

  // Fetch from your own data storage and with your own code
  // the associated application ID and API key for this user
  val appID = getAppIDFor("user42")
  val apiKey = getIndexingApiKeyFor("user42")

  val client = SearchClient(appID, apiKey)
  val searchParams = SearchParamsObject(
    query = "<YOUR_SEARCH_QUERY>",
    facetFilters = FacetFilters.of(
      listOf(
        FacetFilters.of("user:user42"),
        FacetFilters.of("user:public"),
      ),
    ),
  )

  client.searchSingleIndex(indexName = "<YOUR_INDEX_NAME>", searchParams = searchParams)
}
